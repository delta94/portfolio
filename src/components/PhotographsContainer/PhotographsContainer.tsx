import clsx from 'clsx';
import React, { useContext } from 'react';
import { createUseStyles } from 'react-jss';
import { useMediaQuery } from 'react-responsive';
import { Button, Photo } from '..';
import { InformationsContext } from '../../App';
import { CustomTheme } from '../../theme';
import { Informations, PhotoType } from '../../utils/informations';
import { ThreeDAnimation } from '../AnimatedContainer/AnimatedContainer';

interface PhotographsContainerProps {}

const useStyles = createUseStyles((theme: CustomTheme) => ({
  mainContainer: {
    backgroundColor: theme.colors.darkerPrimaryColor,
    color: theme.colors.white,
    display: 'flex',
    padding: theme.spacing(2.5),
    borderRadius: 10,
    height: '100%',
    border: `1px solid ${theme.colors.secondaryColor}`,
    width: '92%',
    flexDirection: 'column',
    textAlign: 'center',
  },
  mainContainerMobile: {
    width: '92%',
  },
  photosContainer: {
    width: '100%',
    display: 'flex',
    flexDirection: 'row',
    flexWrap: 'wrap',
    justifyContent: 'space-evenly',
  },
  button: {
    position: 'relative',
  },
}));

const PhotographsContainer: React.FC<PhotographsContainerProps> = () => {
  const classes: Record<'mainContainer' | 'mainContainerMobile' | 'photosContainer' | 'button', string> = useStyles();
  const isMobile: boolean = useMediaQuery({ maxWidth: 780 });
  const context: Informations | undefined = useContext<Informations | undefined>(InformationsContext);

  return (
    <div className={clsx([classes.mainContainer, isMobile && classes.mainContainerMobile])}>
      <div className={classes.photosContainer}>
        {context?.photos &&
          context.photos.map((photo: PhotoType) => {
            return <ThreeDAnimation key={photo.url}>{<Photo url={photo.url} />}</ThreeDAnimation>;
          })}
      </div>

      <div className={classes.button}>
        <Button onClick={() => window.open('https://www.instagram.com/sk_izsk/', '_blank')}>Explore more images</Button>
      </div>
    </div>
  );
};

export { PhotographsContainer };
