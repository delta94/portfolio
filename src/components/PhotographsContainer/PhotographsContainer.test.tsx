import { render } from '@testing-library/react';
import React from 'react';
import { TestContextProvider } from '../../TestContextProvider';
import { PhotographsContainer } from './PhotographsContainer';

test('PhotographsContainer renders correctly', () => {
  const { asFragment } = render(
    <TestContextProvider>
      <PhotographsContainer />
    </TestContextProvider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
