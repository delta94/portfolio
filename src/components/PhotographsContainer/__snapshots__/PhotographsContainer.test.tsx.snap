// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotographsContainer renders correctly 1`] = `
<DocumentFragment>
  <div
    class="mainContainer-0-2-1"
  >
    <div
      class="photosContainer-0-2-3"
    >
      <div
        style="transform: perspective(600px) rotateX(0deg) rotateY(0deg) scale(1);"
      >
        <img
          alt="photoBanner"
          class="photo-0-2-5"
          src="https://lh3.googleusercontent.com/pw/ACtC-3fn3fohPamP-S-zK3KcM922JgG3qK-MpoGUQyKxGMa4fiSHBWE7GocRT4TEQTQ2RJ-os9-5PBZ9HsOUH9WA82DwhAFK4pJvtZax-03UaLVDZFmQKZNlfr2WzP__mfdzhzikyJp0vokt82o62rCQ05mfKA=w1391-h927-no"
        />
      </div>
      <div
        style="transform: perspective(600px) rotateX(0deg) rotateY(0deg) scale(1);"
      >
        <img
          alt="photoBanner"
          class="photo-0-2-5"
          src="https://lh3.googleusercontent.com/pw/ACtC-3eQQqwZSCuzi8WmWIpCtNxxeWPKgRFY3W8xxDHKTfR2dvxO_xACqMWjkbZdGMrw8AFIpPrniAefxLdvQuljOeKV0U5tmssgwkr8Z27rjNuH7UJ8x0P7yfGRIluskInKtwwQoxx6YPsMyqRQuZdiVdAGog=w1391-h927-no"
        />
      </div>
      <div
        style="transform: perspective(600px) rotateX(0deg) rotateY(0deg) scale(1);"
      >
        <img
          alt="photoBanner"
          class="photo-0-2-5"
          src="https://lh3.googleusercontent.com/pw/ACtC-3fvoWyn4aTnVPqsc4nyVyiSxV4peiPbQGlNPvTXNkdBgYJ7KTqCUhtmrUtzQw60geTDJC-hiyPqX_uh8uAuhhXAtIrigS48BXBvr_LSo1S16qqtF1CX4_g_CqjopQdlwKQGhMpFqZ5YNFAMRxJJI0Ueag=w1391-h927-no"
        />
      </div>
      <div
        style="transform: perspective(600px) rotateX(0deg) rotateY(0deg) scale(1);"
      >
        <img
          alt="photoBanner"
          class="photo-0-2-5"
          src="https://lh3.googleusercontent.com/pw/ACtC-3cX6nKa7XhUbbYnwTeWvZa6VovsMIVzF8GltehNe91O07bnKb9AnuxCpAJ3lHSvfR8T7mPde5WOu3UKbftvBOuSayjAhMd2ZwDKl_f5dAyVl0O8jgQgmHPA-AswABMQEF8V4eaBIaj3Oc5vOpU-13Adbw=w1391-h927-no"
        />
      </div>
      <div
        style="transform: perspective(600px) rotateX(0deg) rotateY(0deg) scale(1);"
      >
        <img
          alt="photoBanner"
          class="photo-0-2-5"
          src="https://lh3.googleusercontent.com/pw/ACtC-3fK0Axi7ncyuRm63A8nrehtXoSIXabywMfhUnyWuErfzZF8r7ZcJhx6Gjc5FcDPlH89xBPtosNEdHBZhkn8M2Jj3_E3rTyJaK8x1KRP539tExiAm2fMLF23OIJkzLqLdu8Me_gPNohLXlW_cnebUVco4A=w1391-h927-no"
        />
      </div>
      <div
        style="transform: perspective(600px) rotateX(0deg) rotateY(0deg) scale(1);"
      >
        <img
          alt="photoBanner"
          class="photo-0-2-5"
          src="https://lh3.googleusercontent.com/pw/ACtC-3dRZQoh-SlCSZGzl-1g2tsTZOK7HCjsNQ97x12lT4eeOTkg0Xt49hoPAwYq9SSx56Fq2YqlFlume637PIwxDHyC9hMrrk7bHwe5VBlw6rQoeMV_BeQGbCxns4u7YdfC2ZFjIY9un3CjElVQLwl_WvSGJQ=w836-h927-no"
        />
      </div>
      <div
        style="transform: perspective(600px) rotateX(0deg) rotateY(0deg) scale(1);"
      >
        <img
          alt="photoBanner"
          class="photo-0-2-5"
          src="https://lh3.googleusercontent.com/pw/ACtC-3ebjvTFu1RTieS3Ao7LJkJLkVraeXYdjhfpHmQAlP9BEt_oEef64Dn7Aj7jq8qt5HkGhvdmjcFn68Hj1MFCCKShbEGIyH-K3AnZh6MuKX9sz0C1UKJBuvXlgfR0tci2LKa2fnjD95vLBB1EFjKT0rwQ1Q=w1391-h927-no"
        />
      </div>
      <div
        style="transform: perspective(600px) rotateX(0deg) rotateY(0deg) scale(1);"
      >
        <img
          alt="photoBanner"
          class="photo-0-2-5"
          src="https://lh3.googleusercontent.com/pw/ACtC-3ci7pPtJtX1mJ471oCJBKOxDIW7K4VzJQmoZbmXoztG5n9RY7AP51m1aCPn-nXY9tOYPr_wL52BBu3HwQ20DuvItkkU_yPgGAqEv7X0fzZJjO5qiD5ZilYU2kCAnf1JJrUT8CCj3N7u2PEC1kOLuvMU7Q=w1391-h927-no"
        />
      </div>
    </div>
    <div
      class="button-0-2-4"
    >
      <button
        class="mainContainer-0-2-8"
        style="transform: perspective(600px) rotateX(0deg) rotateY(0deg) scale(1);"
      >
        Explore more images
      </button>
    </div>
  </div>
</DocumentFragment>
`;
